File: solutions/0049-group-anagrams/solution.java
class Solution {
   public static List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();

        for (String str : strs) {
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String sorted = new String(chars);

            map.computeIfAbsent(sorted, k -> new ArrayList<>()).add(str);
        }

        return new ArrayList<>(map.values());
    }
}

File: solutions/0049-group-anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        vector<vector<string>> res;
        for(int i = 0;i<strs.size();i++){
            string s = strs[i];
            sort(s.begin(), s.end());
            if(mp.find(s)==mp.end()) mp.insert({s, {strs[i]}});
            else {
                mp[s].push_back(strs[i]);
            }
        }
        for (auto &it : mp) {
            vector<string> grp = it.second;
            res.push_back(grp);
        }
        return res;
    }
};
